### ISSUE TITLE
there is no possible to get loss from epoch_metrics

### ISSUE THREAD

**Issue by igor-yusupov** [catalyst-team/catalyst#1162]
## 🐛 Bug Report
In the previous version it was possible to receive information about the loss for the epoch

### How To Reproduce
Run this code sample and you will see that runner.epoch_metrics doesn't have info about loss.

#### Code sample

```
import os
import math
import torch
from catalyst import dl
from catalyst.core.callback import Callback, CallbackOrder, CallbackNode

os.environ["CUDA_VISIBLE_DEVICES"] = "2"

BATCH_SIZE = 4

class MyIterableDataset(torch.utils.data.IterableDataset):
    def __init__(self, start, end):
        super(MyIterableDataset).__init__()
        assert end > start, "this example code only works with end >= start"
        self.start = start
        self.end = end
        self.length = end-start

    def __iter__(self):
        return (x for x in [i*torch.ones(2,requires_grad=False) for i in range(self.start, self.end)])

    def __len__(self):
        return self.length

ds = MyIterableDataset(start=3, end=100)
data_loader = torch.utils.data.DataLoader(ds, batch_size=BATCH_SIZE, num_workers=4)

loaders = {"train":data_loader,"valid":data_loader}
dummy_model = torch.nn.Linear(2,1)

optimizer = torch.optim.Adam(dummy_model.parameters(), lr=0.001)
criterion = torch.nn.L1Loss()
target = torch.empty((BATCH_SIZE,1), dtype=torch.long).random_(5).cuda()


class CustomCallback(Callback):
    def __init__(self):
        super().__init__(order=CallbackOrder.external,
                         node=CallbackNode.master)
    
    def on_epoch_end(self, runner: str = "IRunner"):
        print(runner.epoch_metrics)


class CustomRunner(dl.Runner):
   
    def handle_batch(self, batch):
        output = self.model(batch)        
        loss = criterion(output,target[:output.shape[0]])       
        self.batch_metrics.update({"loss": loss.clone().detach()})

        if self.is_train_loader:
            loss.backward()            
            self.optimizer.step()
            self.optimizer.zero_grad()

runner = CustomRunner()

# model training
runner.train(
    model=dummy_model,
    optimizer=optimizer,
    loaders=loaders,
    logdir="../logs",
    num_epochs=2,
    verbose=True,
    callbacks=[CustomCallback()]
)
```

**Comment by Scitator** [comment]
Yeah, that's correct: if you are computing the metric directly in the loop – you have to add them directly to the batch and loader storages. Please follow the docs [here](https://catalyst-team.github.io/catalyst/faq/logging.html#logging).

**Comment by igor-yusupov** [comment]
ok, I got it. thanks!
